### ISSUE TITLE
LDAP Security - Don't re-authenticate on every request

### ISSUE THREAD

**Issue by hyperman1** [quarkusio/quarkus#10267]
**Description**
When using ldap security and servlets, every request/response re-authenticates.  Especially the role lookup can take a lot of time.  Hence, the elytron cache should be possible to enable

**Implementation ideas**
The existing io.quarkus.elytron.security.ldap.LdapRecorder class seems a good point for this:  The last line is: 

          return new RuntimeValue<>(builder.build());

You can do something like:
               //These come from application.properties
               int config_cache_size=10000;
               long config_cache_time_to_live=1000L*60*15;//15 minutes

                SecurityRealm ldapRealm = builder.build();
		if (ldapRealm instanceof CacheableSecurityRealm&&config_cache_size!=0) {
			// Cache 15 minuten
			ldapRealm = new CachingSecurityRealm((CacheableSecurityRealm) ldapRealm,
					new LRURealmIdentityCache(config_cache_size, config_cache_time_to_live));
		}
                return new RuntimeValue<>(ldapRealm);

**Comment by hartimcwildfly** [comment]
Should the caching be fixed at buildtime or at runtime?

**Comment by daniel-chelcioiu** [comment]
Hi, 
Any news on this ? 
Thanks

**Comment by hyperman1** [comment]
@hartimcwildfly  I think I answered you in another thread or on the chat, so I'll add my answer from long ago here:

In my experience,  all security parameters should be tuneable at run time.  Security setup can differ between dev and prod, and might be adapted by different teams.

We managed to introduce keycloak in our organisation, so my interest for LDAP has seriously diminished since writing these calls.

@daniel-chelcioiu  I don't think there is any other news on this.
